components/handlers/module_edit_preclusions.py

```diff
@@ -0,0 +1,46 @@
+'''
+    This module contains the handler for web requests pertaining to
+    the editing of a module's preclusions.
+'''
+
+
+import json
+from app import RENDER
+import web
+from components import model, session
+
+
+class EditModulePreclusions(object):
+    '''
+        This class handles the editing of a module's preclusions.
+    '''
+    def GET(self):
+        '''
+            Handles the loading of the 'Edit Module Preclusions' page.
+        '''
+        if not session.validate_session():
+            raise web.seeother('/login')
+        else:
+            input_data = model.validate_input(web.input(), ["code"])
+            module_code = input_data.code
+
+            module_info = model.get_module(module_code)
+            if module_info is None:
+                raise web.notfound(RENDER.notfound("Module " +\
+                 module_code + " does not exist in the system."))
+
+            preclusions = model.get_preclusion_units(module_code)
+            return RENDER.moduleEditPreclusion(module_code, preclusions)
+
+
+    def POST(self):
+        '''
+            Handles the submission of updated module preclusions
+            for a target module.
+        '''
+        module_code = web.input().get('code')
```

Comment:
Should also use model.validate_input() in the POST method to validate the module code.

FYI, validate_input is able to check: 1) if module code is specified, 2) if module code exists in the system, and 3) ignore upper/lower cases in module codes. (Currently, the edit pre-req/preclusion pages are not accepting lower case module codes)

For POST method, need to set show_404=False to prevent the 404 page from being rendered.
If input is valid, the function will return the input itself, else it will return False.

Can refer to module_edit.py to see how to use validate_input in POST method.